[quant][bc-breaking][improvements] Removed quantized_max_pool1d registration

dzdang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #75305
* #75261
* #75304
* #72353

Summary: This PR is part of a series of PRs addressing https://github.com/pytorch/pytorch/issues/54150,
related to using dispatcher for calls to quantized backends as opposed to if/else conditionals
.
quantized_max_pool1d has been removed from the frontend, which was previously not used anyways, after introducing
max_pool2d_with_indices support for quantized tensors.

BC-breaking note:
Previous:
```
import torch
x = torch.rand(1, 1, 1)
qx = torch.quantize_per_tensor(x, scale=0.5, zero_point=0, dtype=torch.qint8)
max_pooled_qx = torch.quantized_max_pool1d(qx, kernel_size=1)
print(qx)
print(max_pooled_qx)
```

Result:
```
tensor([[[0.5000]]], size=(1, 1, 1), dtype=torch.qint8,
       quantization_scheme=torch.per_tensor_affine, scale=0.5, zero_point=0)
tensor([[[0.5000]]], size=(1, 1, 1), dtype=torch.qint8,
       quantization_scheme=torch.per_tensor_affine, scale=0.5, zero_point=0)
```

After:
```
import torch
x = torch.rand(1, 1, 1, 1)
qx = torch.quantize_per_tensor(x, scale=0.5, zero_point=0, dtype=torch.qint8)
max_pooled_qx = torch.max_pool1d(qx, kernel_size=1)
print(qx)
print(max_pooled_qx)
```

Result:
```
tensor([[[[1.]]]], size=(1, 1, 1, 1), dtype=torch.qint8,
       quantization_scheme=torch.per_tensor_affine, scale=0.5, zero_point=0)
tensor([[[[1.]]]], size=(1, 1, 1, 1), dtype=torch.qint8,
       quantization_scheme=torch.per_tensor_affine, scale=0.5, zero_point=0)
```

Test plan:
```
python test/test_quantization.py -k test_max_pool1d
```

Differential Revision: [D35431830](https://our.internmc.facebook.com/intern/diff/D35431830)

dzdang committed:
[quant][bc-breaking][improvements] Removed quantized_max_pool1d registration

Summary: This PR is part of a series of PRs addressing https://github.com/pytorch/pytorch/issu
es/54150,
related to using dispatcher for calls to quantized backends as opposed to if/else conditionals
.
quantized_max_pool1d has been removed from the frontend, which was previously not used anyways, after introducing
max_pool2d_with_indices support for quantized tensors.

Test plan:
```
python test/test_quantization.py -k test_max_pool1d
```

[ghstack-poisoned]

dzdang committed:
Update on "[quant][bc-breaking][improvements] Removed quantized_max_pool1d registration"


Summary: This PR is part of a series of PRs addressing https://github.com/pytorch/pytorch/issu
es/54150,
related to using dispatcher for calls to quantized backends as opposed to if/else conditionals
.
quantized_max_pool1d has been removed from the frontend, which was previously not used anyways, after introducing
max_pool2d_with_indices support for quantized tensors.

Test plan:
```
python test/test_quantization.py -k test_max_pool1d
```

[ghstack-poisoned]

dzdang committed:
Update on "[quant][bc-breaking][improvements] Removed quantized_max_pool1d registration"


Summary: This PR is part of a series of PRs addressing https://github.com/pytorch/pytorch/issu
es/54150,
related to using dispatcher for calls to quantized backends as opposed to if/else conditionals
.
quantized_max_pool1d has been removed from the frontend, which was previously not used anyways, after introducing
max_pool2d_with_indices support for quantized tensors.

Test plan:
```
python test/test_quantization.py -k test_max_pool1d
```

[ghstack-poisoned]

dzdang committed:
Update on "[quant][bc-breaking][improvements] Removed quantized_max_pool1d registration"


Summary: This PR is part of a series of PRs addressing https://github.com/pytorch/pytorch/issu
es/54150,
related to using dispatcher for calls to quantized backends as opposed to if/else conditionals
.
quantized_max_pool1d has been removed from the frontend, which was previously not used anyways, after introducing
max_pool2d_with_indices support for quantized tensors.

Test plan:
```
python test/test_quantization.py -k test_max_pool1d
```

[ghstack-poisoned]

dzdang committed:
Update on "[quant][bc-breaking][improvements] Removed quantized_max_pool1d registration"


Summary: This PR is part of a series of PRs addressing https://github.com/pytorch/pytorch/issues/54150,
related to using dispatcher for calls to quantized backends as opposed to if/else conditionals
.
quantized_max_pool1d has been removed from the frontend, which was previously not used anyways, after introducing
max_pool2d_with_indices support for quantized tensors.

Test plan:
```
python test/test_quantization.py -k test_max_pool1d
```

[ghstack-poisoned]

dzdang committed:
Update on "[quant][bc-breaking][improvements] Removed quantized_max_pool1d registration"


Summary: This PR is part of a series of PRs addressing https://github.com/pytorch/pytorch/issues/54150,
related to using dispatcher for calls to quantized backends as opposed to if/else conditionals
.
quantized_max_pool1d has been removed from the frontend, which was previously not used anyways, after introducing
max_pool2d_with_indices support for quantized tensors.

Test plan:
```
python test/test_quantization.py -k test_max_pool1d
```

[ghstack-poisoned]

dzdang commented:
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35431830).

dzdang committed:
Update on "[quant][bc-breaking][improvements] Removed quantized_max_pool1d registration"


Summary: This PR is part of a series of PRs addressing https://github.com/pytorch/pytorch/issues/54150,
related to using dispatcher for calls to quantized backends as opposed to if/else conditionals
.
quantized_max_pool1d has been removed from the frontend, which was previously not used anyways, after introducing
max_pool2d_with_indices support for quantized tensors.

Test plan:
```
python test/test_quantization.py -k test_max_pool1d
```

Differential Revision: [D35431830](https://our.internmc.facebook.com/intern/diff/D35431830)

[ghstack-poisoned]

dzdang committed:
Update on "[quant][bc-breaking][improvements] Removed quantized_max_pool1d registration"


Summary: This PR is part of a series of PRs addressing https://github.com/pytorch/pytorch/issues/54150,
related to using dispatcher for calls to quantized backends as opposed to if/else conditionals
.
quantized_max_pool1d has been removed from the frontend, which was previously not used anyways, after introducing
max_pool2d_with_indices support for quantized tensors.

Test plan:
```
python test/test_quantization.py -k test_max_pool1d
```

Differential Revision: [D35431830](https://our.internmc.facebook.com/intern/diff/D35431830)

[ghstack-poisoned]

dzdang commented:
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35431830).

jerryzh168 commented:
same for this one, we need to have bc-breaking notes: https://github.com/pytorch/pytorch/pull/70009

dzdang committed:
Update on "[quant][bc-breaking][improvements] Removed quantized_max_pool1d registration"


Summary: This PR is part of a series of PRs addressing https://github.com/pytorch/pytorch/issues/54150,
related to using dispatcher for calls to quantized backends as opposed to if/else conditionals
.
quantized_max_pool1d has been removed from the frontend, which was previously not used anyways, after introducing
max_pool2d_with_indices support for quantized tensors.

BC-breaking note:
Previous:
```
import torch
x = torch.rand(1, 1, 1)
qx = torch.quantize_per_tensor(x, scale=0.5, zero_point=0, dtype=torch.qint8)
max_pooled_qx = torch.quantized_max_pool1d(qx, kernel_size=1)
print(qx)
print(max_pooled_qx)
```

Result:
```
tensor([[[0.5000]]], size=(1, 1, 1), dtype=torch.qint8,
       quantization_scheme=torch.per_tensor_affine, scale=0.5, zero_point=0)
tensor([[[0.5000]]], size=(1, 1, 1), dtype=torch.qint8,
       quantization_scheme=torch.per_tensor_affine, scale=0.5, zero_point=0)
```

After:
```
import torch
x = torch.rand(1, 1, 1)
qx = torch.quantize_per_tensor(x, scale=0.5, zero_point=0, dtype=torch.qint8)
max_pooled_qx = torch.quantized_max_pool1d(qx, kernel_size=1)
print(qx)
print(max_pooled_qx)
```

Result
```
AttributeError: module 'torch' has no attribute 'quantized_max_pool1d'
```

Now we must use:
```
import torch
x = torch.rand(1, 1, 1, 1)
qx = torch.quantize_per_tensor(x, scale=0.5, zero_point=0, dtype=torch.qint8)
max_pooled_qx = torch.max_pool1d(qx, kernel_size=1)
print(qx)
print(max_pooled_qx)
```

Result
```
tensor([[[[1.]]]], size=(1, 1, 1, 1), dtype=torch.qint8,
       quantization_scheme=torch.per_tensor_affine, scale=0.5, zero_point=0)
tensor([[[[1.]]]], size=(1, 1, 1, 1), dtype=torch.qint8,
       quantization_scheme=torch.per_tensor_affine, scale=0.5, zero_point=0)
```

Test plan:
```
python test/test_quantization.py -k test_max_pool1d
```

Differential Revision: [D35431830](https://our.internmc.facebook.com/intern/diff/D35431830)

[ghstack-poisoned]

dzdang commented:
> same for this one, we need to have bc-breaking notes: #70009

@jerryzh168 added. can you approv?

dzdang committed:
Update on "[quant][bc-breaking][improvements] Removed quantized_max_pool1d registration"


Summary: This PR is part of a series of PRs addressing https://github.com/pytorch/pytorch/issues/54150,
related to using dispatcher for calls to quantized backends as opposed to if/else conditionals
.
quantized_max_pool1d has been removed from the frontend, which was previously not used anyways, after introducing
max_pool2d_with_indices support for quantized tensors.

BC-breaking note:
Previous:
```
import torch
x = torch.rand(1, 1, 1)
qx = torch.quantize_per_tensor(x, scale=0.5, zero_point=0, dtype=torch.qint8)
max_pooled_qx = torch.quantized_max_pool1d(qx, kernel_size=1)
print(qx)
print(max_pooled_qx)
```

Result:
```
tensor([[[0.5000]]], size=(1, 1, 1), dtype=torch.qint8,
       quantization_scheme=torch.per_tensor_affine, scale=0.5, zero_point=0)
tensor([[[0.5000]]], size=(1, 1, 1), dtype=torch.qint8,
       quantization_scheme=torch.per_tensor_affine, scale=0.5, zero_point=0)
```

After:
```
import torch
x = torch.rand(1, 1, 1)
qx = torch.quantize_per_tensor(x, scale=0.5, zero_point=0, dtype=torch.qint8)
max_pooled_qx = torch.quantized_max_pool1d(qx, kernel_size=1)
print(qx)
print(max_pooled_qx)
```

Result
```
AttributeError: module 'torch' has no attribute 'quantized_max_pool1d'
```

Now we must use:
```
import torch
x = torch.rand(1, 1, 1, 1)
qx = torch.quantize_per_tensor(x, scale=0.5, zero_point=0, dtype=torch.qint8)
max_pooled_qx = torch.max_pool1d(qx, kernel_size=1)
print(qx)
print(max_pooled_qx)
```

Result
```
tensor([[[[1.]]]], size=(1, 1, 1, 1), dtype=torch.qint8,
       quantization_scheme=torch.per_tensor_affine, scale=0.5, zero_point=0)
tensor([[[[1.]]]], size=(1, 1, 1, 1), dtype=torch.qint8,
       quantization_scheme=torch.per_tensor_affine, scale=0.5, zero_point=0)
```

Test plan:
```
python test/test_quantization.py -k test_max_pool1d
```

Differential Revision: [D35431830](https://our.internmc.facebook.com/intern/diff/D35431830)

[ghstack-poisoned]

dzdang commented:
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35431830).

dzdang committed:
Update on "[quant][bc-breaking][improvements] Removed quantized_max_pool1d registration"


Summary: This PR is part of a series of PRs addressing https://github.com/pytorch/pytorch/issues/54150,
related to using dispatcher for calls to quantized backends as opposed to if/else conditionals
.
quantized_max_pool1d has been removed from the frontend, which was previously not used anyways, after introducing
max_pool2d_with_indices support for quantized tensors.

BC-breaking note:
Previous:
```
import torch
x = torch.rand(1, 1, 1)
qx = torch.quantize_per_tensor(x, scale=0.5, zero_point=0, dtype=torch.qint8)
max_pooled_qx = torch.quantized_max_pool1d(qx, kernel_size=1)
print(qx)
print(max_pooled_qx)
```

Result:
```
tensor([[[0.5000]]], size=(1, 1, 1), dtype=torch.qint8,
       quantization_scheme=torch.per_tensor_affine, scale=0.5, zero_point=0)
tensor([[[0.5000]]], size=(1, 1, 1), dtype=torch.qint8,
       quantization_scheme=torch.per_tensor_affine, scale=0.5, zero_point=0)
```

After:
```
import torch
x = torch.rand(1, 1, 1)
qx = torch.quantize_per_tensor(x, scale=0.5, zero_point=0, dtype=torch.qint8)
max_pooled_qx = torch.quantized_max_pool1d(qx, kernel_size=1)
print(qx)
print(max_pooled_qx)
```

Result
```
AttributeError: module 'torch' has no attribute 'quantized_max_pool1d'
```

Now we must use:
```
import torch
x = torch.rand(1, 1, 1, 1)
qx = torch.quantize_per_tensor(x, scale=0.5, zero_point=0, dtype=torch.qint8)
max_pooled_qx = torch.max_pool1d(qx, kernel_size=1)
print(qx)
print(max_pooled_qx)
```

Result
```
tensor([[[[1.]]]], size=(1, 1, 1, 1), dtype=torch.qint8,
       quantization_scheme=torch.per_tensor_affine, scale=0.5, zero_point=0)
tensor([[[[1.]]]], size=(1, 1, 1, 1), dtype=torch.qint8,
       quantization_scheme=torch.per_tensor_affine, scale=0.5, zero_point=0)
```

Test plan:
```
python test/test_quantization.py -k test_max_pool1d
```

Differential Revision: [D35431830](https://our.internmc.facebook.com/intern/diff/D35431830)

[ghstack-poisoned]

dzdang commented:
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35431830).

dzdang committed:
Update on "[quant][bc-breaking][improvements] Removed quantized_max_pool1d registration"


Summary: This PR is part of a series of PRs addressing https://github.com/pytorch/pytorch/issues/54150,
related to using dispatcher for calls to quantized backends as opposed to if/else conditionals
.
quantized_max_pool1d has been removed from the frontend, which was previously not used anyways, after introducing
max_pool2d_with_indices support for quantized tensors.

BC-breaking note:
Previous:
```
import torch
x = torch.rand(1, 1, 1)
qx = torch.quantize_per_tensor(x, scale=0.5, zero_point=0, dtype=torch.qint8)
max_pooled_qx = torch.quantized_max_pool1d(qx, kernel_size=1)
print(qx)
print(max_pooled_qx)
```

Result:
```
tensor([[[0.5000]]], size=(1, 1, 1), dtype=torch.qint8,
       quantization_scheme=torch.per_tensor_affine, scale=0.5, zero_point=0)
tensor([[[0.5000]]], size=(1, 1, 1), dtype=torch.qint8,
       quantization_scheme=torch.per_tensor_affine, scale=0.5, zero_point=0)
```

After:
```
import torch
x = torch.rand(1, 1, 1)
qx = torch.quantize_per_tensor(x, scale=0.5, zero_point=0, dtype=torch.qint8)
max_pooled_qx = torch.quantized_max_pool1d(qx, kernel_size=1)
print(qx)
print(max_pooled_qx)
```

Result
```
AttributeError: module 'torch' has no attribute 'quantized_max_pool1d'
```

Now we must use:
```
import torch
x = torch.rand(1, 1, 1, 1)
qx = torch.quantize_per_tensor(x, scale=0.5, zero_point=0, dtype=torch.qint8)
max_pooled_qx = torch.max_pool1d(qx, kernel_size=1)
print(qx)
print(max_pooled_qx)
```

Result
```
tensor([[[[1.]]]], size=(1, 1, 1, 1), dtype=torch.qint8,
       quantization_scheme=torch.per_tensor_affine, scale=0.5, zero_point=0)
tensor([[[[1.]]]], size=(1, 1, 1, 1), dtype=torch.qint8,
       quantization_scheme=torch.per_tensor_affine, scale=0.5, zero_point=0)
```

Test plan:
```
python test/test_quantization.py -k test_max_pool1d
```

Differential Revision: [D35431830](https://our.internmc.facebook.com/intern/diff/D35431830)

[ghstack-poisoned]

dzdang commented:
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35431830).

dzdang committed:
Update on "[quant][bc-breaking][improvements] Removed quantized_max_pool1d registration"


Summary: This PR is part of a series of PRs addressing https://github.com/pytorch/pytorch/issues/54150,
related to using dispatcher for calls to quantized backends as opposed to if/else conditionals
.
quantized_max_pool1d has been removed from the frontend, which was previously not used anyways, after introducing
max_pool2d_with_indices support for quantized tensors.

BC-breaking note:
Previous:
```
import torch
x = torch.rand(1, 1, 1)
qx = torch.quantize_per_tensor(x, scale=0.5, zero_point=0, dtype=torch.qint8)
max_pooled_qx = torch.quantized_max_pool1d(qx, kernel_size=1)
print(qx)
print(max_pooled_qx)
```

Result:
```
tensor([[[0.5000]]], size=(1, 1, 1), dtype=torch.qint8,
       quantization_scheme=torch.per_tensor_affine, scale=0.5, zero_point=0)
tensor([[[0.5000]]], size=(1, 1, 1), dtype=torch.qint8,
       quantization_scheme=torch.per_tensor_affine, scale=0.5, zero_point=0)
```

After:
```
import torch
x = torch.rand(1, 1, 1)
qx = torch.quantize_per_tensor(x, scale=0.5, zero_point=0, dtype=torch.qint8)
max_pooled_qx = torch.quantized_max_pool1d(qx, kernel_size=1)
print(qx)
print(max_pooled_qx)
```

Result
```
AttributeError: module 'torch' has no attribute 'quantized_max_pool1d'
```

Now we must use:
```
import torch
x = torch.rand(1, 1, 1, 1)
qx = torch.quantize_per_tensor(x, scale=0.5, zero_point=0, dtype=torch.qint8)
max_pooled_qx = torch.max_pool1d(qx, kernel_size=1)
print(qx)
print(max_pooled_qx)
```

Result
```
tensor([[[[1.]]]], size=(1, 1, 1, 1), dtype=torch.qint8,
       quantization_scheme=torch.per_tensor_affine, scale=0.5, zero_point=0)
tensor([[[[1.]]]], size=(1, 1, 1, 1), dtype=torch.qint8,
       quantization_scheme=torch.per_tensor_affine, scale=0.5, zero_point=0)
```

Test plan:
```
python test/test_quantization.py -k test_max_pool1d
```

Differential Revision: [D35431830](https://our.internmc.facebook.com/intern/diff/D35431830)

[ghstack-poisoned]

dzdang commented:
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35431830).

dzdang committed:
Update on "[quant][bc-breaking][improvements] Removed quantized_max_pool1d registration"


Summary: This PR is part of a series of PRs addressing https://github.com/pytorch/pytorch/issues/54150,
related to using dispatcher for calls to quantized backends as opposed to if/else conditionals
.
quantized_max_pool1d has been removed from the frontend, which was previously not used anyways, after introducing
max_pool2d_with_indices support for quantized tensors.

BC-breaking note:
Previous:
```
import torch
x = torch.rand(1, 1, 1)
qx = torch.quantize_per_tensor(x, scale=0.5, zero_point=0, dtype=torch.qint8)
max_pooled_qx = torch.quantized_max_pool1d(qx, kernel_size=1)
print(qx)
print(max_pooled_qx)
```

Result:
```
tensor([[[0.5000]]], size=(1, 1, 1), dtype=torch.qint8,
       quantization_scheme=torch.per_tensor_affine, scale=0.5, zero_point=0)
tensor([[[0.5000]]], size=(1, 1, 1), dtype=torch.qint8,
       quantization_scheme=torch.per_tensor_affine, scale=0.5, zero_point=0)
```

After:
```
import torch
x = torch.rand(1, 1, 1, 1)
qx = torch.quantize_per_tensor(x, scale=0.5, zero_point=0, dtype=torch.qint8)
max_pooled_qx = torch.max_pool1d(qx, kernel_size=1)
print(qx)
print(max_pooled_qx)
```

Result:
```
tensor([[[[1.]]]], size=(1, 1, 1, 1), dtype=torch.qint8,
       quantization_scheme=torch.per_tensor_affine, scale=0.5, zero_point=0)
tensor([[[[1.]]]], size=(1, 1, 1, 1), dtype=torch.qint8,
       quantization_scheme=torch.per_tensor_affine, scale=0.5, zero_point=0)
```

Test plan:
```
python test/test_quantization.py -k test_max_pool1d
```

Differential Revision: [D35431830](https://our.internmc.facebook.com/intern/diff/D35431830)

[ghstack-poisoned]

dzdang committed:
Update on "[quant][bc-breaking][improvements] Removed quantized_max_pool1d registration"


Summary: This PR is part of a series of PRs addressing https://github.com/pytorch/pytorch/issues/54150,
related to using dispatcher for calls to quantized backends as opposed to if/else conditionals
.
quantized_max_pool1d has been removed from the frontend, which was previously not used anyways, after introducing
max_pool2d_with_indices support for quantized tensors.

BC-breaking note:
Previous:
```
import torch
x = torch.rand(1, 1, 1)
qx = torch.quantize_per_tensor(x, scale=0.5, zero_point=0, dtype=torch.qint8)
max_pooled_qx = torch.quantized_max_pool1d(qx, kernel_size=1)
print(qx)
print(max_pooled_qx)
```

Result:
```
tensor([[[0.5000]]], size=(1, 1, 1), dtype=torch.qint8,
       quantization_scheme=torch.per_tensor_affine, scale=0.5, zero_point=0)
tensor([[[0.5000]]], size=(1, 1, 1), dtype=torch.qint8,
       quantization_scheme=torch.per_tensor_affine, scale=0.5, zero_point=0)
```

After:
```
import torch
x = torch.rand(1, 1, 1, 1)
qx = torch.quantize_per_tensor(x, scale=0.5, zero_point=0, dtype=torch.qint8)
max_pooled_qx = torch.max_pool1d(qx, kernel_size=1)
print(qx)
print(max_pooled_qx)
```

Result:
```
tensor([[[[1.]]]], size=(1, 1, 1, 1), dtype=torch.qint8,
       quantization_scheme=torch.per_tensor_affine, scale=0.5, zero_point=0)
tensor([[[[1.]]]], size=(1, 1, 1, 1), dtype=torch.qint8,
       quantization_scheme=torch.per_tensor_affine, scale=0.5, zero_point=0)
```

Test plan:
```
python test/test_quantization.py -k test_max_pool1d
```

Differential Revision: [D35431830](https://our.internmc.facebook.com/intern/diff/D35431830)

[ghstack-poisoned]

dzdang committed:
Update on "[quant][bc-breaking][improvements] Removed quantized_max_pool1d registration"


Summary: This PR is part of a series of PRs addressing https://github.com/pytorch/pytorch/issues/54150,
related to using dispatcher for calls to quantized backends as opposed to if/else conditionals
.
quantized_max_pool1d has been removed from the frontend, which was previously not used anyways, after introducing
max_pool2d_with_indices support for quantized tensors.

BC-breaking note:
Previous:
```
import torch
x = torch.rand(1, 1, 1)
qx = torch.quantize_per_tensor(x, scale=0.5, zero_point=0, dtype=torch.qint8)
max_pooled_qx = torch.quantized_max_pool1d(qx, kernel_size=1)
print(qx)
print(max_pooled_qx)
```

Result:
```
tensor([[[0.5000]]], size=(1, 1, 1), dtype=torch.qint8,
       quantization_scheme=torch.per_tensor_affine, scale=0.5, zero_point=0)
tensor([[[0.5000]]], size=(1, 1, 1), dtype=torch.qint8,
       quantization_scheme=torch.per_tensor_affine, scale=0.5, zero_point=0)
```

After:
```
import torch
x = torch.rand(1, 1, 1, 1)
qx = torch.quantize_per_tensor(x, scale=0.5, zero_point=0, dtype=torch.qint8)
max_pooled_qx = torch.max_pool1d(qx, kernel_size=1)
print(qx)
print(max_pooled_qx)
```

Result:
```
tensor([[[[1.]]]], size=(1, 1, 1, 1), dtype=torch.qint8,
       quantization_scheme=torch.per_tensor_affine, scale=0.5, zero_point=0)
tensor([[[[1.]]]], size=(1, 1, 1, 1), dtype=torch.qint8,
       quantization_scheme=torch.per_tensor_affine, scale=0.5, zero_point=0)
```

Test plan:
```
python test/test_quantization.py -k test_max_pool1d
```

Differential Revision: [D35431830](https://our.internmc.facebook.com/intern/diff/D35431830)

[ghstack-poisoned]

dzdang committed:
Update on "[quant][bc-breaking][improvements] Removed quantized_max_pool1d registration"


Summary: This PR is part of a series of PRs addressing https://github.com/pytorch/pytorch/issues/54150,
related to using dispatcher for calls to quantized backends as opposed to if/else conditionals
.
quantized_max_pool1d has been removed from the frontend, which was previously not used anyways, after introducing
max_pool2d_with_indices support for quantized tensors.

BC-breaking note:
Previous:
```
import torch
x = torch.rand(1, 1, 1)
qx = torch.quantize_per_tensor(x, scale=0.5, zero_point=0, dtype=torch.qint8)
max_pooled_qx = torch.quantized_max_pool1d(qx, kernel_size=1)
print(qx)
print(max_pooled_qx)
```

Result:
```
tensor([[[0.5000]]], size=(1, 1, 1), dtype=torch.qint8,
       quantization_scheme=torch.per_tensor_affine, scale=0.5, zero_point=0)
tensor([[[0.5000]]], size=(1, 1, 1), dtype=torch.qint8,
       quantization_scheme=torch.per_tensor_affine, scale=0.5, zero_point=0)
```

After:
```
import torch
x = torch.rand(1, 1, 1, 1)
qx = torch.quantize_per_tensor(x, scale=0.5, zero_point=0, dtype=torch.qint8)
max_pooled_qx = torch.max_pool1d(qx, kernel_size=1)
print(qx)
print(max_pooled_qx)
```

Result:
```
tensor([[[[1.]]]], size=(1, 1, 1, 1), dtype=torch.qint8,
       quantization_scheme=torch.per_tensor_affine, scale=0.5, zero_point=0)
tensor([[[[1.]]]], size=(1, 1, 1, 1), dtype=torch.qint8,
       quantization_scheme=torch.per_tensor_affine, scale=0.5, zero_point=0)
```

Test plan:
```
python test/test_quantization.py -k test_max_pool1d
```

Differential Revision: [D35431830](https://our.internmc.facebook.com/intern/diff/D35431830)

[ghstack-poisoned]

dzdang commented:
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35431830).
